src/wallet/wallet.cpp

```diff
@@ -4098,6 +4099,38 @@ bool CWallet::BackupWallet(const std::string& strDest)
     return dbw->Backup(strDest);
 }
 
+bool CWallet::PendingNameFirstUpdateExists(const std::string &name)
+{
+    LOCK(cs_wallet);
+    return namePendingMap.end() != namePendingMap.find(name);
+}
+
+bool CWallet::WritePendingNameFirstUpdate(const std::string &name, const std::string &rand, const std::string &txid, const std::string &data, const std::string &toaddress)
+{
+    LOCK(cs_wallet);
+    CNamePendingData nameData;
+    nameData.setHex(txid);
+    nameData.setRand(rand);
+    nameData.setData(data);
+    nameData.setToAddress(toaddress);
+    return CWalletDB(*dbw).WriteNameFirstUpdate(name, nameData);
+}
+
+bool CWallet::ErasePendingNameFirstUpdate(const std::string &name)
+{
+    LOCK(cs_wallet);
+    namePendingMap.erase(name);
+    return CWalletDB(*dbw).EraseNameFirstUpdate(name);
+}
+
+CNamePendingData CWallet::GetPendingNameFirstUpdate(const std::string &name)
+{
+    LOCK(cs_wallet);
+    std::map<std::string, CNamePendingData> pn = namePendingMap;
+    std::map<std::string, CNamePendingData>::iterator it = pn.find(name);
+    return it->second;
```

Comment:
How can your caller know if the name was not found?  (Since `namePendingMap.end()` is presumably private, if you have this access function defined.)